### PR TITLE
Integrate symbol error

### PR DESCRIPTION
**Pull request by antonrohr**
This changes the symbol loading error handling by integrating the
SymbolErrorDialog. In the SymbolErrorDialog the user can either cancel
the symbol loading, click try again, or go to the SymbolsDialog (Symbol
Locations). In the later to cases, Orbit tries to load the symbols
again.

http://b/153637024

Avoid duplicate symbol loading attempts

http://b/220095606

### CODE REVIEW

File: src/OrbitQt/orbitmainwindow.cpp
Comment (reichlfl) [inline src/OrbitQt/orbitmainwindow.cpp:364]: This is a nitpick, but I think the `return` in two cases and `break` in one makes the flow of this hard to read. I would find it clearer to structure this like so:

```
if (result == orbit_config_widgets::SymbolErrorDialog::Result::kAddSymbolLocation) {
  ...
  return true;
}

return result == orbit_config_widgets::SymbolErrorDialog::Result::kTryAgain;
```

Alternatively, move the code after the switch into a separate method and call it inside the `switch` statement.

**Review by reichlfl — APPROVED** [review]
Minor comments, LGTM. I would wait for an additional approval, I'm not sure if I have a grasp on potential side effects of this.

File: src/OrbitQt/orbitmainwindow.cpp
Comment (antonrohr) [reply src/OrbitQt/orbitmainwindow.cpp:364]: Hmm. I actually disagree. 
1. I do like enum handling in a switch a lot. Especially without a default case, because it result in a compiler warning when someone in the future adds a value to the enum. 
2. I would prefer early returns over moving the `...` part into an if, but I guess thats just personal preference.
3. I dont like the `return result == orbit_config_widgets::SymbolErrorDialog::Result::kTryAgain;`. Maybe I am not used to thinking about boolean logic in that way, but figuring when this returns true and when false, is more cumbersome to me than reading ifs. (Now that I write this and think about it, I actually get the idea and it grows a bit on me).  

I like the switch, because I read it like this:  There is a switch, all cases are handled. The `kCancel` case result in an immediate return of false. The `kTryAgain` results in a return of true and in the `kAddSymbolLocation` case there is more things that follow afterwards. The order of the cases helps here.. first early exit and otherwise more things afterwards.  
wdyt?

Comment (antonrohr) [reply src/OrbitQt/orbitmainwindow.cpp:364]: > Alternatively, move the code after the switch into a separate method and call it inside the switch statement.

I forget to read this.. I also like it.

Comment (reichlfl) [reply src/OrbitQt/orbitmainwindow.cpp:364]: I get the points, and I agree about the enum handling in a switch. But keep in mind that someone might not want to read your method carefully line by line, but just skim over it to get the general idea, and I think it's easy to be missed that one of the switch cases behaves differently.

Granted, in this case it's not a big deal because you probably immediately wonder why there's code after a switch that returns, so you'll read more properly and realize this.

I am in general not a fan of early returns in a method because of the reason above, it's easy to miss. I much more prefer the an approach where there should be only one exit point of a method, except for error case handling, and all the error case handling happens at the top. But I am clearly a minority in the team, so I won't complain and adapt ;-)

Comment (antonrohr) [reply src/OrbitQt/orbitmainwindow.cpp:364]: > I guess the separate method is the best of both worlds.

Yeah I kinda decided on that a minute after I responded.. There is also an opportunity to dedup with some other code.

File: src/OrbitGl/App.cpp
Comment (antonrohr) [reply src/OrbitGl/App.cpp:1696]: done

File: src/OrbitGl/App.h
Comment (antonrohr) [reply src/OrbitGl/App.h:274]: done

File: src/OrbitQt/orbitmainwindow.cpp
Comment (antonrohr) [reply src/OrbitQt/orbitmainwindow.cpp:364]: done

**Pull request merged**
